src/lib.rs

```diff
@@ -948,16 +952,56 @@ impl<S: VhostUserBackend> VhostUserSlaveReqHandlerMut for VhostUserHandler<S> {
 
     fn get_inflight_fd(
         &mut self,
-        _inflight: &vhost::vhost_user::message::VhostUserInflight,
-    ) -> VhostUserResult<(vhost::vhost_user::message::VhostUserInflight, File)> {
-        // Assume the backend hasn't negotiated the inflight feature; it
-        // wouldn't be correct for the backend to do so, as we don't (yet)
-        // provide a way for it to handle such requests.
-        Err(VhostUserError::InvalidOperation)
+        inflight: &VhostUserInflight,
+    ) -> VhostUserResult<(VhostUserInflight, File)> {
+        let inflight_queue_region = unsafe {
+            libc::syscall(
+                libc::SYS_memfd_create,
+                &std::ffi::CString::new("inflight-region").unwrap(),
+                libc::MFD_CLOEXEC,
+            )
+        };
+
```

Comment:
Did you use `vu_get_inflight_fd()` from https://github.com/qemu/qemu/blob/master/subprojects/libvhost-user/libvhost-user.c#L1654 as a reference?
From what I see, you forgot to call into `fcntl(*fd, F_ADD_SEALS, flags)` in order to add `F_SEAL_GROW`, `F_SEAL_SHRINK` and `F_SEAL_SEAL` to setup the proper seals, and you didn't `mmap()` the fd into the `vhost-user` process address space.